preferencesfx/src/main/java/com/dlsc/preferencesfx/util/StorageHandler.java

```diff
@@ -160,7 +160,7 @@ public double loadWindowPosY() {
    */
   // asciidoctor Documentation - tag::storageHandlerSave[]
   public void saveObject(String breadcrumb, Object object) {
-    preferences.put(hash(breadcrumb), gson.toJson(object));
+    preferences.put(breadcrumb, gson.toJson(object));
```

Comment:
Hashing the breadcrumb is intentional, have a look at the hash() method lines 217-225 for the explanation of why we use it.
We agree it's not a very elegant solution to this problem, but if you for example run the extended demo and change the following setting: "Appearance & Behaviour#UI Options#Adjust colors for red-green vision defiency (protanopia, deuteranopia)", you will notice an exception being thrown upon saving (in your branch). We think being able to save settings with longer names with the trade-off of having non-readable keys in the stored preferences is a worthwile trade-off.
But I do realize that the issue is that your listener doesn't work anymore if this is hashed... **Maybe you could add a logger entry here and in loadObject, logging the breadcrumb as a string before it is hashed?** It would have about the same effect, while still preserving the possibility of saving longer keys.